From f86726945b821939a2d4620094f4f5f0a866a564 Mon Sep 17 00:00:00 2001
From: Jeremy Newton <Jeremy.Newton@amd.com>
Date: Mon, 31 Jan 2022 17:54:14 -0500
Subject: [PATCH 1/3] SWDEV-321118 - Use GNUInstallDirs

Use GNUInstallDirs variables to determine the location of BINDIR,
LIBDIR, DOCDIR, and SYSCONFDIR.

Signed-off-by: Jeremy Newton <Jeremy.Newton@amd.com>
Change-Id: Ifb38263dce80d80c5ebb4eacd8e49c76bb013a44
---
 CMakeLists.txt                    |  2 +-
 packaging/CMakeLists.txt          | 12 ++++++------
 packaging/rocm-ocl-icd.postinst   |  4 ++--
 packaging/rocm-ocl-icd.prerm      |  4 ++--
 packaging/rocm-ocl-icd.rpm_post   |  4 ++--
 packaging/rocm-ocl-icd.rpm_postun |  4 ++--
 6 files changed, 15 insertions(+), 15 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5c4537..0612fa2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ if (POLICY CMP0048)
   set(PROJ_VERSION VERSION 1.5.0)
 endif()
 
-project(opencl)
+project(rocm-opencl)
 
 include(GNUInstallDirs)
 
diff --git a/packaging/CMakeLists.txt b/packaging/CMakeLists.txt
index 13a2185..2b6fa77 100644
--- a/packaging/CMakeLists.txt
+++ b/packaging/CMakeLists.txt
@@ -1,13 +1,13 @@
 cmake_minimum_required(VERSION 3.5.1)
-project(rocm-opencl)
+include(GNUInstallDirs)
 
 set(CPACK_COMPONENTS_ALL binary dev icd)
 set(CPACK_DEB_COMPONENT_INSTALL ON)
 set(CPACK_RPM_COMPONENT_INSTALL ON)
 
-install(TARGETS clinfo DESTINATION bin COMPONENT binary)
-install(TARGETS amdocl DESTINATION lib COMPONENT binary)
-install(FILES ${CMAKE_SOURCE_DIR}/LICENSE.txt DESTINATION share/doc/${CMAKE_PROJECT_NAME} COMPONENT binary)
+install(TARGETS clinfo DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT binary)
+install(TARGETS amdocl DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT binary)
+install(FILES ${CMAKE_SOURCE_DIR}/LICENSE.txt DESTINATION ${CMAKE_INSTALL_DOCDIR} COMPONENT binary)
 
 install(DIRECTORY ${CMAKE_SOURCE_DIR}/khronos/headers/opencl2.2/CL
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} COMPONENT dev
@@ -17,8 +17,8 @@ install(DIRECTORY ${CMAKE_SOURCE_DIR}/khronos/headers/opencl2.2/CL
         PATTERN cl_dx9_media_sharing.h EXCLUDE
         PATTERN cl_egl.h EXCLUDE )
 
-install(TARGETS OpenCL DESTINATION lib COMPONENT icd )
-install(FILES ${CMAKE_SOURCE_DIR}/khronos/icd/LICENSE DESTINATION share/doc/rocm-ocl-icd COMPONENT icd)
+install(TARGETS OpenCL DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT icd )
+install(FILES ${CMAKE_SOURCE_DIR}/khronos/icd/LICENSE DESTINATION ${CMAKE_INSTALL_DATADIR}/doc/rocm-ocl-icd COMPONENT icd)
 
 
 # Generic CPACK variables
diff --git a/packaging/rocm-ocl-icd.postinst b/packaging/rocm-ocl-icd.postinst
index 0780970..7c46ade 100644
--- a/packaging/rocm-ocl-icd.postinst
+++ b/packaging/rocm-ocl-icd.postinst
@@ -6,8 +6,8 @@ INSTALL_PATH=@CPACK_PACKAGING_INSTALL_PREFIX@
 ROCM_LIBPATH=@ROCM_PATH@/lib
 
 do_ldconfig() {
-  echo ${INSTALL_PATH}/lib > /etc/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
-  mkdir -p /etc/OpenCL/vendors && (echo libamdocl64.so > /etc/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@)
+  echo ${INSTALL_PATH}/@CMAKE_INSTALL_LIBDIR@ > /@CMAKE_INSTALL_SYSCONFDIR@/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
+  mkdir -p /@CMAKE_INSTALL_SYSCONFDIR@/OpenCL/vendors && (echo libamdocl64.so > /@CMAKE_INSTALL_SYSCONFDIR@/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@)
 }
 
 case "$1" in
diff --git a/packaging/rocm-ocl-icd.prerm b/packaging/rocm-ocl-icd.prerm
index 9fc6c8d..e0b5417 100644
--- a/packaging/rocm-ocl-icd.prerm
+++ b/packaging/rocm-ocl-icd.prerm
@@ -3,8 +3,8 @@
 set -e
 
 rm_ldconfig() {
-  rm -f /etc/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
-  rm -f /etc/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@
+  rm -f /@CMAKE_INSTALL_SYSCONFDIR@/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
+  rm -f /@CMAKE_INSTALL_SYSCONFDIR@/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@
 }
 
 case "$1" in
diff --git a/packaging/rocm-ocl-icd.rpm_post b/packaging/rocm-ocl-icd.rpm_post
index 7ce5d97..49129ce 100644
--- a/packaging/rocm-ocl-icd.rpm_post
+++ b/packaging/rocm-ocl-icd.rpm_post
@@ -1,8 +1,8 @@
 INSTALL_PATH=@CPACK_PACKAGING_INSTALL_PREFIX@
 ROCM_LIBPATH=@ROCM_PATH@/lib
 
-echo ${INSTALL_PATH}/lib > /etc/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
-mkdir -p /etc/OpenCL/vendors && (echo libamdocl64.so > /etc/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@)
+echo ${INSTALL_PATH}/@CMAKE_INSTALL_LIBDIR@ > /@CMAKE_INSTALL_SYSCONFDIR@/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
+mkdir -p /@CMAKE_INSTALL_SYSCONFDIR@/OpenCL/vendors && (echo libamdocl64.so > /@CMAKE_INSTALL_SYSCONFDIR@/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@)
 
 mkdir -p ${ROCM_LIBPATH}
 ln -s -f -r ${INSTALL_PATH}/lib/libOpenCL.so ${ROCM_LIBPATH}/libOpenCL.so
diff --git a/packaging/rocm-ocl-icd.rpm_postun b/packaging/rocm-ocl-icd.rpm_postun
index 356c7af..44bbce3 100644
--- a/packaging/rocm-ocl-icd.rpm_postun
+++ b/packaging/rocm-ocl-icd.rpm_postun
@@ -1,6 +1,6 @@
 if [ $1 -eq 0 ]; then
-  rm -f /etc/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
-  rm -f /etc/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@
+  rm -f /@CMAKE_INSTALL_SYSCONFDIR@/ld.so.conf.d/10-rocm-opencl.conf && ldconfig
+  rm -f /@CMAKE_INSTALL_SYSCONFDIR@/OpenCL/vendors/@OPENCL_AMD_ICD_FILE@
   rm -f @ROCM_PATH@/lib/libOpenCL.so
   rm -f @ROCM_PATH@/lib/libOpenCL.so.@OPENCL_LIB_VERSION_MAJOR@
   rm -f @ROCM_PATH@/lib/libOpenCL.so.@OPENCL_LIB_VERSION_STRING@
-- 
2.35.1

